apps/web/components/dashboard-components/project-details/custom-llm-parameters-editor.tsx

```diff
@@ -0,0 +1,650 @@
+            initial={{ opacity: 0, y: 10 }}
+            animate={{ opacity: 1, y: 0 }}
+            exit={{ opacity: 0, y: -10 }}
+            transition={{ duration: 0.3, ease: "easeInOut" }}
+            className="space-y-4"
+          >
+            <CardDescription className="text-sm text-foreground max-w-sm mb-4">
+              Add custom parameters to send with each LLM request. These will be
+              passed as provider options.
+            </CardDescription>
+
+            {/* Suggestions for common parameters */}
+            {suggestions.length > 0 && (
+              <div className="mb-4">
+                <Label className="text-xs text-muted-foreground mb-2 block">
+                  Common parameters for {providerName}:
+                </Label>
+                <div className="flex flex-wrap gap-2">
+                  {suggestions.map((suggestion) => (
+                    <Button
+                      key={suggestion.key}
+                      variant="outline"
+                      size="sm"
+                      onClick={() => applySuggestion(suggestion)}
+                      className="text-xs"
+                      title={suggestion.description}
+                    >
+                      <Plus className="h-3 w-3 mr-1" />
+                      {suggestion.key}
+                    </Button>
+                  ))}
+                </div>
+              </div>
+            )}
+
+            {/* Parameter list */}
+            {parameters.length > 0 ? (
+              <div className="space-y-3">
+                {parameters.map((param, index) => (
+                  <motion.div
+                    key={index}
+                    initial={{ opacity: 0, x: -10 }}
+                    animate={{ opacity: 1, x: 0 }}
+                    exit={{ opacity: 0, x: 10 }}
+                    className="flex gap-2 items-start"
+                  >
+                    <Input
+                      placeholder="Parameter name"
+                      value={param.key}
+                      onChange={(e) =>
+                        updateParameter(index, "key", e.target.value)
+                      }
+                      className="flex-1"
+                    />
+                    <Select
+                      value={param.type}
+                      onValueChange={(value) =>
+                        updateParameter(index, "type", value)
+                      }
+                    >
+                      <SelectTrigger className="w-[120px]">
+                        <SelectValue />
+                      </SelectTrigger>
+                      <SelectContent>
+                        <SelectItem value="string">String</SelectItem>
+                        <SelectItem value="number">Number</SelectItem>
+                        <SelectItem value="boolean">Boolean</SelectItem>
+                      </SelectContent>
+                    </Select>
+                    {param.type === "boolean" ? (
```

Comment:
having complex inline ternaries in JSX tends to make it really hard to edit these files. Can you just break out stuff like this into separate subcomponents?

(I think look for any places in this PR that have `) : (` as a sign of something that should be factored better)